airbyte-webapp/src/hooks/services/ConnectionEdit/ConnectionEditService.tsx

```diff
@@ -11,22 +17,69 @@ interface ConnectionEditProps {
   connectionId: string;
 }
 
-const useConnectionEdit = ({ connectionId }: ConnectionEditProps) => {
-  const [connection, setConnection] = useState(useGetConnection(connectionId));
+export interface ConnectionCatalog {
+  syncCatalog: AirbyteCatalog;
+  catalogId?: string;
+}
+
+interface ConnectionEditHook {
+  connection: WebBackendConnectionRead;
+  connectionUpdating: boolean;
+  schemaError?: Error;
+  schemaRefreshing: boolean;
+  schemaHasBeenRefreshed: boolean;
+  updateConnection: (connectionUpdates: WebBackendConnectionUpdate) => Promise<void>;
+  refreshSchema: () => Promise<void>;
+  discardRefreshedSchema: () => void;
+}
+
+const getConnectionCatalog = (connection: WebBackendConnectionRead): ConnectionCatalog =>
+  pick(connection, ["syncCatalog", "catalogId"]);
```

Comment:
Has there been any discussion or chatter around removing lodash? I'm fine using it but it's a chunky package so maybe it'd be better to just pull these values out?